src/api/artifact/controller.go

```diff
@@ -32,17 +36,19 @@ type Controller interface {
 	// creates it if it doesn't exist. If tags are provided, ensure they exist
 	// and are attached to the artifact. If the tags don't exist, create them first
 	Ensure(ctx context.Context, repository *models.RepoRecord, digest string, tags ...string) (err error)
-	// List artifacts according to the query and option
+	// List artifacts according to the query, specify the properties returned with option
 	List(ctx context.Context, query *q.Query, option *Option) (total int64, artifacts []*Artifact, err error)
 	// Get the artifact specified by ID
 	Get(ctx context.Context, id int64, option *Option) (artifact *Artifact, err error)
 	// Delete the artifact specified by ID
 	Delete(ctx context.Context, id int64) (err error)
-	// DeleteTag deletes the tag specified by ID
-	DeleteTag(ctx context.Context, id int64) (err error)
-	// UpdatePullTime updates the pull time for the artifact. If the tag is provides, update the pull
+	// Tags returns the tags according to the query, specify the properties returned with option
+	Tags(ctx context.Context, query *q.Query, option *TagOption) (total int64, tags []*Tag, err error)
```

Comment:
Is it possible to return a list of tags of different artifacts?

Reply:
It is. If you want to list the tags of one specific artifact, specify the query

Reply:
I was thinking there should be a restriction but as long as it's by design I'm OK.  Feel free to resolve this one.

Reply:
Restricting the tags belong to only one artifact makes no sense, I think listing all tags under one specific repository is a reasonable requirement, and this is needed for `/v2/{name}/tags/list` either